docs/services/OneDrive.md

```diff
@@ -116,106 +100,97 @@ while (OneDriveItems != null);
+### Creating or uploading files - that exceed 4MB
 
 ```csharp
 
-// If the file exceed the Maximum size (ie 4MB) use the UploadFileAsync method instead
-var largeFileCreated = await folder.UploadFileAsync(selectedFile.Name, localStream, CreationCollisionOption.GenerateUniqueName, 320 * 1024);
-
-```
-
-### Moving, copying and renaming files
-
-```csharp
-
-// You can also Move, Copy or Rename a file
-await fileCreated.MoveAsync(destFolder);
-await fileCreated.CopyAsync(destFolder);
-await fileCreated.RenameAsync("newName");
+var selectedFile = await OpenLocalFileAsync();
+if (selectedFile != null)
+    {
+        using (var localStream = await selectedFile.OpenReadAsync())
+        {
+            Shell.Current.DisplayWaitRing = true;
+
+            // If the file exceed the Maximum size (ie 4MB)
+            var largeFileCreated = await folder.StorageFolderPlatformService.UploadFileAsync(selectedFile.Name, localStream, CreationCollisionOption.GenerateUniqueName, 320 * 1024);
+        }
+    }
+}
 
 ```
-
 ### Downloading files
 
 ```csharp
 
 // Download a file and save the content in a local file
-var remoteFile=await level3Folder.GetFile("NewFile.docx"); 
-
-using (var remoteStream = await remoteFile.OpenAsync())
- {
-     byte[] buffer = new byte[remoteStream.Size];
-     var localBuffer = await remoteStream.ReadAsync(buffer.AsBuffer(), (uint)remoteStream.Size, InputStreamOptions.ReadAhead);
-	 var localFolder = ApplicationData.Current.LocalFolder;
-     var myLocalFile = await localFolder.CreateFileAsync($"{oneDriveFile.Name}", CreationCollisionOption.GenerateUniqueName);
-     using (var localStream = await myLocalFile.OpenAsync(FileAccessMode.ReadWrite))
-     {
-         await localStream.WriteAsync(localBuffer);
-         await localStream.FlushAsync();
-     }
- }
+// Convert the storage item to a storage file
+var oneDriveFile = (Toolkit.Services.OneDrive.OneDriveStorageFile)item;
+using (var remoteStream = (await oneDriveFile.StorageFilePlatformService.OpenAsync()) as IRandomAccessStream)
+{
+    // Use a helper method to open local filestream and write to it 
+    await SaveToLocalFolder(remoteStream, oneDriveFile.Name);
+}
 
 ```
 
 ### Retrieving file thumbnails
 
 ```csharp
 
-// At last you can get the thumbnail of a file
-var stream = await file.GetThumbnailAsync(ThumbnailSize.Large)
-Windows.UI.Xaml.Controls.Image thumbnail = new Windows.UI.Xaml.Controls.Image();
-BitmapImage bmp = new BitmapImage();
-await bmp.SetSourceAsync(streamTodDisplay);
-thumbnail.Source = bmp;
+var file = (Toolkit.Services.OneDrive.OneDriveStorageItem)((AppBarButton)e.OriginalSource).DataContext;
+using (var stream = (await file.StorageItemPlatformService.GetThumbnailAsync(Toolkit.Services.MicrosoftGraph.MicrosoftGraphEnums.ThumbnailSize.Large)) as IRandomAccessStream)
+{
+    // Use a helper method to display the images on the xaml view
+    await OneDriveSampleHelpers.DisplayThumbnail(stream, "thumbnail");
+}
 
 ```
   
```

Comment:
In the *Related Topics* section it'd be useful to have a link to the underlying OneDrive graph api docs.  See the new [doc template](https://github.com/Microsoft/UWPCommunityToolkit/blob/master/docs/.template.md).

Reply:
Will add

Reply:
Done